### PR TITLE
Fix: Fix bugs for v3.15

### PR DESCRIPTION
**Pull request by meiliang1123**
Set proper camera position of Canvas.toFront when machine changed
Fix duplicate action in laser/cnc
Show days of elapsed time
Disable model actions of context menu when support being selected
Do not remove special chars of backend files
Fix SVG Flip not effected on toolpath in laser & cnc
Clear toolpath when opening new file
Set Save-As dialog as modal of current window
Fix supports disapear when doing undo
Disable rotate when multi-selecting supports
Fix gcode file name overflow problem

### CODE REVIEW

**Comment by meiliang1123** [comment]
Reference Internal JIRA: SB-130 SB-237 SB-201 SB-260 SB-261

File: src/app/flux/editor/index.js [hunk 1]
```diff
@@ -537,13 +537,14 @@ export const actions = {
...
 
     duplicateSelectedModel: (headType) => (dispatch, getState) => {
-        const { page, modelGroup, toolPathModelGroup } = getState()[headType];
+        const { page, modelGroup } = getState()[headType];
         if (page === PAGE_PROCESS) return;
 
         const { originalName, uploadName, config, sourceType, sourceWidth, sourceHeight, mode, transformation } = modelGroup.getSelectedModel();
-        const { gcodeConfig } = toolPathModelGroup.getSelectedModel();
+        transformation.positionX += 5;
+        transformation.positionY -= 5;
```
Comment (bbaa77770) [inline src/app/flux/editor/index.js:545]: why?

**Review by bbaa77770 — APPROVED** [review]
LGTM

**Review by zjiefee — APPROVED** [review]

**Review by jane-rose — APPROVED** [review]

File: src/app/flux/editor/index.js (hunk 1 above)
Comment (meiliang1123) [reply src/app/flux/editor/index.js:545]: add an offset for the new element

**Pull request merged**